opm/utility/ECLCaseUtilities.hpp

```diff
@@ -0,0 +1,56 @@
+/*
+  Copyright 2017 Statoil ASA.
+
+  This file is part of the Open Porous Media Project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef OPM_ECLCASEUTILITIES_HEADER_INCLUDED
+#define OPM_ECLCASEUTILITIES_HEADER_INCLUDED
+
+#include <vector>
+
+#include <boost/filesystem/path.hpp>
+
+namespace Opm { namespace ECLCaseUtilities {
+
+    class ResultSet
+    {
+    public:
+        using Path = boost::filesystem::path;
+
+        explicit ResultSet(const Path& casename);
+
+        Path gridFile() const;
+        Path initFile() const;
+        Path restartFile(const int reportStepID) const;
+
+        bool isUnifiedRestart() const;
+
+        std::vector<int> reportStepIDs() const;
```

Comment:
The class is self-explanatory and does not require much in the way of added docs. I just wonder: will reportStepIDs() typically start with 0 or 1?

Reply:
> will reportStepIDs() typically start with 0 or 1?

Yes :wink:

That actually depends on how the simulation run is configured.  If the case requests that the initial state be output, then it will start at zero.  Otherwise it will (usually) start at one.